webapp/src/utils/versionUtils.ts

```
@@ -0,0 +1,144 @@
+/**
+ * Copyright (c) 2025, RTE (https://www.rte-france.com)
+ *
+ * See AUTHORS.txt
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ * SPDX-License-Identifier: MPL-2.0
+ *
+ * This file is part of the Antares project.
+ */
+
+import debug from "debug";
+import semver from "semver";
+import { isNumericValue } from "./numberUtils";
+
+const log = debug("antares:utils:versionUtils");
+
+export const ZERO_SEMVER = "0.0.0" as const;
+
+// The maximum value for any semver component is `Number.MAX_SAFE_INTEGER`
+export const MAX_SEMVER =
+  `${Number.MAX_SAFE_INTEGER}.${Number.MAX_SAFE_INTEGER}.${Number.MAX_SAFE_INTEGER}` as const;
+
+/**
+ * Normalizes a version value from the API into a full semantic version string (MAJOR.MINOR.PATCH).
+ *
+ * Supported input formats:
+ * - 3-digit numeric (e.g. 930 or "930")  -> "9.3.0"
+ * - Partial string (e.g. "9.3" -> "9.3.0", "9" -> "9.0.0")
+ * - Any string accepted by `semver.coerce()`
+ *
+ * This normalization ensures the result can be safely used with semver library.
+ *
+ * @param version - Version as number or string.
+ * @returns A semantic version string, or "0.0.0" on failure.
+ */
+export function toSemanticVersion(version: number | string) {
+  const numberVerStr = version.toString();
+
+  // e.g., 930 or "930" -> "9.3.0"
+  if (numberVerStr.length === 3 && numberVerStr.split("").every(isNumericValue)) {
+    return `${numberVerStr[0]}.${numberVerStr[1]}.${numberVerStr[2]}`;
+  }
+
+  const semVer = semver.valid(semver.coerce(version));
+
+  // e.g., "9.3" -> "9.3.0" or "9" -> "9.0.0"
+  if (semVer === null) {
+    log("toSemanticVersion(): invalid version value '%s'", version);
+    return ZERO_SEMVER;
```

is this on purpose to silently return "0.0.0" on failure ? you're not informing the caller about the error, and displaying potential wrong versions if I am not wrong